API/Services/Implementations/ScreeningService.cs

```diff
@@ -0,0 +1,125 @@
+using API.DTOs;
+using API.Enums;
+using API.Services.Interfaces;
+
+namespace API.Services.Implementations;
+
+public class ScreeningService(IPlayerService playerService, IBaseStatsService baseStatsService, IPlayerStatsService
+    playerStatsService) : IScreeningService
+{
+    public async Task<ScreeningResultDTO> ScreenAsync(ScreeningRequestDTO screeningRequest)
+    {
+        var idList = screeningRequest.OsuPlayerIds.ToList();
+
+        if (idList.Count == 0)
+        {
+            throw new Exception("Screening id list cannot be empty");
+        }
+
+        var passed = 0;
+        var failed = 0;
+
+        var resultCollection = new List<PlayerScreeningResultDTO>();
+        foreach (var osuId in idList)
+        {
+            PlayerInfoDTO? playerInfo = await playerService.GetAsync(osuId);
+            (ScreeningResult result, ScreeningFailReason? failReason) = await ScreenAsync(screeningRequest, playerInfo);
+
+            switch (result)
+            {
+                case ScreeningResult.Pass:
+                    passed++;
+                    break;
+                case ScreeningResult.Fail:
+                    failed++;
+                    break;
+                default:
+                    throw new Exception("This ScreeningResult is not handled!");
+            }
+
+            resultCollection.Add(new PlayerScreeningResultDTO
+            {
+                PlayerId = playerInfo?.Id,
+                Username = playerInfo?.Username,
+                OsuPlayerId = osuId,
+                ScreeningResult = result,
+                ScreeningFailReason = failReason
+            });
+        }
+
+        var dto = new ScreeningResultDTO
+        {
+            PlayersPassed = passed,
+            PlayersFailed = failed,
+            ScreeningResults = resultCollection
+        };
+
+        return dto;
+    }
+
+    private async Task<(ScreeningResult result, ScreeningFailReason? failReason)> ScreenAsync(ScreeningRequestDTO screeningRequest,
+        PlayerInfoDTO? playerInfo)
+    {
+        ScreeningResult result = ScreeningResult.Pass;
+        ScreeningFailReason? failReason = ScreeningFailReason.None;
+
+        if (playerInfo == null)
+        {
+            result = ScreeningResult.Fail;
+            failReason |= ScreeningFailReason.NoData;
+
+            return (result, failReason);
+        }
+
+
+        BaseStatsDTO? baseStats = await baseStatsService.GetAsync(null, playerInfo.Id, screeningRequest.Ruleset);
+        var peakRating = await playerStatsService.GetPeakRatingAsync(playerInfo.Id, screeningRequest.Ruleset);
```

Comment:
Can be defined all the way at the bottom where it is actually used. Sticking it above all of these return cases is redundant.